java/main/src/main/java/com/saucelabs/saucebindings/VisualSession.java

```diff
@@ -0,0 +1,85 @@
+package com.saucelabs.saucebindings;
+
+import com.saucelabs.saucebindings.options.SauceOptions;
+import com.saucelabs.saucebindings.options.VisualOptions;
+import org.openqa.selenium.InvalidArgumentException;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class VisualSession extends SauceSession {
+    private VisualResults visualResults;
+    private final VisualOptions visualOptions;
+
+    public VisualSession(String testName) {
+        this(new VisualOptions(SauceOptions.chrome().setName(testName).build()));
+    }
+
+    public VisualSession(VisualOptions options) {
+        super(options.getSauceOptions());
+        this.visualOptions = options;
+    }
+
+    @Override
+    public RemoteWebDriver start() {
+        this.driver = createRemoteWebDriver(getSauceUrl(), visualOptions.toCapabilities());
+        newVisualTest(getSauceOptions().sauce().getName());
+        return driver;
+    }
+
+    @Override
+    public URL getSauceUrl() {
+        try {
+            return new URL("https://hub.screener.io/wd/hub");
```

Comment:
should the URL be moved into a constant/config?

Reply:
Good point. the `getSauceUrl()` in the superclass is based on `sauceUrl` field, which is a Data Center Enum, which doesn't apply for visual. How we use it here is probably going to depend on if we're planning on having a new visual data center or somehow merge the screener.io into saucelabs.com. I'll get more info.

Reply:
This is what I just got about future data center URL, so I should definitely create a `VisualDataCenter` enum for these and update when necessary.

> [11:19 AM] the future endpoints are going to be [visual.eu-central-1.saucelabs.com](http://visual.eu-central-1.saucelabs.com/) and [visual.hub.eu-central-1.saucelabs.com](http://visual.hub.eu-central-1.saucelabs.com/)